src/backend/utils/mmgr/aset.c

```diff
@@ -998,6 +1005,104 @@ static void AllocSetReleaseAccountingForAllAllocatedChunks(MemoryContext context
 #endif
 }
 
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
```

Comment:
I still think we can move this and other same parts to separate files keeping a few public objects like AllocSetGetAllocatedChunkStats(), MemoryContext_printTopListOfChunks() and only a few structures.
But it's debatable.